project/ParadoxSupport.scala

```diff
@@ -0,0 +1,98 @@
+        val updatedNode = new DirectiveNode(DirectiveNode.Format.Inline, node.name, label, source, attributes, null,  new TextNode(label))
+        renderByFqcn(updatedNode, visitor, printer)
+      }
+      else {
+        renderByClassName(node.label, node, visitor, printer)
+      }
+    }
+
+    def renderByClassName(label: String, node: DirectiveNode, visitor: Visitor, printer: Printer): Unit = {
+      def syntheticNode(group: String, label: String, c: String): DirectiveNode = {
+        val syntheticSource = new DirectiveNode.Source.Direct(c)
+        val attributes = new org.pegdown.ast.DirectiveAttributes.AttributeMap()
+        new DirectiveNode(DirectiveNode.Format.Inline, group, null, null, attributes, null,
+          new DirectiveNode(DirectiveNode.Format.Inline, group + "doc", label, syntheticSource, node.attributes, c,
+            new TextNode(label)
+          ))
+      }
+
+      val scalaLabel = label.replaceAll("\\\\_", "_")
+      val labelWithoutGenericParameters = scalaLabel.split("\\[")(0)
+      val javaLabel = scalaLabel.replaceAll("\\[", "&lt;").replaceAll("\\]", "&gt;").replace('_', '?')
+      val matches = allClasses.filter(_.endsWith('.' + labelWithoutGenericParameters))
+      matches.size match {
+        case 0 =>
+          throw new java.lang.IllegalStateException(s"No matches found for $label.")
+        case 1 if matches(0).contains("adsl") =>
+          throw new java.lang.IllegalStateException(s"Match for $label only found in one language: ${matches(0)}")
+        case 1 =>
+          syntheticNode("scala", scalaLabel, matches(0)).accept(visitor)
+          syntheticNode("java", javaLabel, matches(0)).accept(visitor)
+        case 2 if matches.forall(_.contains("adsl")) =>
+          matches.foreach(m => {
+            if (!m.contains("javadsl"))
+              syntheticNode("scala", scalaLabel, m).accept(visitor)
+            if (!m.contains("scaladsl"))
+              syntheticNode("java", javaLabel, m).accept(visitor)
+          })
+        case n =>
+          throw new java.lang.IllegalStateException(
+            s"$n matches found for $label, but not javadsl/scaladsl: ${matches.mkString(", ")}. " +
+              s"You may want to use the fully qualified class name as @unidoc[fqcn] instead of @unidoc[${label}]."
+          )
+      }
+    }
+
+    def renderByFqcn(node: DirectiveNode, visitor: Visitor, printer: Printer): Unit = {
+      try{
+        printer.print(s"""<span class="group-scala">""")
+        ScaladocDirective(page, variables).render(node, visitor, printer)
+        printer.print(s"</span>")
+
+        printer.print(s"""<span class="group-java">""")
+        JavadocDirective(page, variables).render(node, visitor, printer)
+        printer.print(s"</span>")
+      } catch {
+        case e: LinkException =>
+          throw new LinkException(s"${e.getMessage}. You may want to use the fully qualified class name as `@unidoc[fqcn]`.")
```

Comment:
Error message is like:

> [error] Caused by: com.lightbend.paradox.markdown.ExternalLinkDirective$LinkException: Failed to resolve [client.ClusterReceptionist] referenced from [cluster-client.html] because property [javadoc.base_url] is not defined. You may want to use the fully qualified class name as @unidoc[fqcn].
>[error]         at akka.ParadoxSupport$UnidocDirective.renderByFqcn(ParadoxSupport.scala:94)

Reply:
Nice!